File: dugout-api/src/main/kotlin/com/dugout/api/domain/team/entity/Team.kt
package com.dugout.api.domain.team.entity

import com.dugout.api.global.common.BaseEntity
import com.dugout.api.global.common.StringListConverter
import jakarta.persistence.Column
import jakarta.persistence.Convert
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.Table

@Entity
@Table(name = "teams")
class Team(
    @Column(nullable = false, length = 100)
    var name: String,

    @Column(name = "logo_url", length = 500)
    var logoUrl: String? = null,

    @Column(nullable = false, length = 50)
    var region: String,

    @Column(nullable = false)
    var division: Int = 4,

    @Column(name = "home_ground_id")
    var homeGroundId: Long? = null,

    @Convert(converter = StringListConverter::class)
    @Column(name = "activity_days", length = 200, nullable = false)
    var activityDays: List<String> = emptyList(),

    @Column(name = "activity_time", length = 20)
    var activityTime: String? = null,

    @Column(name = "invite_code", unique = true, length = 20)
    var inviteCode: String? = null,

    @Enumerated(EnumType.STRING)
    @Column(name = "lineup_mode", nullable = false, length = 20)
    var lineupMode: LineupMode = LineupMode.BALANCED,
) : BaseEntity() {

    fun update(
        name: String?,
        logoUrl: String?,
        region: String?,
        division: Int?,
        activityDays: List<String>?,
        activityTime: String?,
        lineupMode: LineupMode?,
    ) {
        name?.let { this.name = it }
        logoUrl?.let { this.logoUrl = it }
        region?.let { this.region = it }
        division?.let { this.division = it }
        activityDays?.let { this.activityDays = it }
        activityTime?.let { this.activityTime = it }
        lineupMode?.let { this.lineupMode = it }
    }

    fun updateInviteCode(code: String) {
        this.inviteCode = code
    }

    companion object {
        fun create(
            name: String,
            region: String,
            division: Int = 4,
            logoUrl: String? = null,
            activityDays: List<String> = emptyList(),
            activityTime: String? = null,
            lineupMode: LineupMode = LineupMode.BALANCED,
        ): Team = Team(
            name = name,
            region = region,
            division = division,
            logoUrl = logoUrl,
            activityDays = activityDays,
            activityTime = activityTime,
            lineupMode = lineupMode,
        )
    }
}

File: dugout-ios/Features/Team/Sources/Domain/Entities/Team.swift
//
//  Team.swift
//  DugoutTeamFeature
//

import Foundation

/// 팀 상세 Entity.
public struct Team: Sendable, Equatable, Identifiable, Hashable {
    public let id: Int64
    public let name: String
    public let logoUrl: String?
    public let region: String
    public let division: Int
    public let homeGroundId: Int64?
    public let activityDays: [String]
    public let activityTime: String?
    public let inviteCode: String?
    public let lineupMode: LineupMode
    public let memberCount: Int
    public let createdAt: Date

    public init(
        id: Int64,
        name: String,
        logoUrl: String?,
        region: String,
        division: Int,
        homeGroundId: Int64?,
        activityDays: [String],
        activityTime: String?,
        inviteCode: String?,
        lineupMode: LineupMode,
        memberCount: Int,
        createdAt: Date
    ) {
        self.id = id
        self.name = name
        self.logoUrl = logoUrl
        self.region = region
        self.division = division
        self.homeGroundId = homeGroundId
        self.activityDays = activityDays
        self.activityTime = activityTime
        self.inviteCode = inviteCode
        self.lineupMode = lineupMode
        self.memberCount = memberCount
        self.createdAt = createdAt
    }
}

public enum LineupMode: String, Sendable, Hashable {
    case balanced = "BALANCED"
    case competitive = "COMPETITIVE"
}
